File: server/src/test/java/com/example/everylive/EveryLiveServerApplicationTest.java
package com.example.everylive;

import static org.assertj.core.api.Assertions.assertThat;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.web.server.LocalServerPort;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class EveryLiveServerApplicationTest {

    private final HttpClient httpClient = HttpClient.newHttpClient();

    @LocalServerPort
    private int port;

    @Test
    void rootEndpointReturnsServerStatus() throws Exception {
        String body = get("/");

        assertThat(body).isEqualTo("EveryLive Spring Boot server is running");
    }

    @Test
    void healthEndpointReturnsUp() throws Exception {
        String body = get("/health");

        assertThat(body).contains("\"status\":\"UP\"");
    }

    private String get(String path) throws Exception {
        HttpRequest request = HttpRequest.newBuilder(URI.create("http://localhost:" + port + path))
                .GET()
                .build();

        HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());

        assertThat(response.statusCode()).isEqualTo(200);
        return response.body();
    }
}

File: server/src/test/kotlin/com/example/everylive/EveryLiveServerApplicationTest.kt
package com.example.everylive

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.web.client.TestRestTemplate
import org.springframework.boot.test.web.server.LocalServerPort

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
class EveryLiveServerApplicationTest {
    @LocalServerPort
    private var port: Int = 0

    @Autowired
    private lateinit var restTemplate: TestRestTemplate

    @Test
    fun rootEndpointReturnsServerStatus() {
        val body = restTemplate.getForObject("http://localhost:$port/", String::class.java)

        assertEquals("EveryLive Spring Boot server is running", body)
    }

    @Test
    fun healthEndpointReturnsUp() {
        val body = restTemplate.getForObject("http://localhost:$port/health", Map::class.java)

        assertEquals("UP", body["status"])
    }
}
